calm_adapter/window_generator/window_generator.py

```diff
@@ -0,0 +1,71 @@
+#!/bin/env python3
+
+from datetime import date, timedelta
+import click
+import json
+import boto3
+from tqdm import tqdm
+
+
+class WindowGenerator:
+
+    topic_arn = "arn:aws:sns:eu-west-1:760097843905:calm-adapter"
+    role_arn = "arn:aws:iam::760097843905:role/platform-developer"
+
+    def __init__(self, start, end=None):
+        self.start = start
+        self.end = end or start
+        assert self.start <= self.end, "Start window not be after end window"
+        self.client = self.sns_client
+
+    def run(self):
+        for window in tqdm([*self.windows]):
```

Comment:
This is slightly odd. Are you trying to work out how many windows there will be? If so, `list(self.windows)` might be a bit clearer.

Reply:
I prefer this syntax for "unravelling" a generator into a list personally, but happy to change it